lms/djangoapps/courseware/views/views.py

```diff
@@ -1583,6 +1613,17 @@ def render_xblock(request, usage_key_string, check_if_enrolled=True):
                     )
                 )
 
+        # For courses using the edx-exams service:
+        # Access to exam content is determined by edx-exams and passed the the LMS using a
+        # signed JWT. There is no longer a need for exam gating or logic inside the
+        # sequence block or its render call. As a result decendants_are_gated will not return
+        # true for these exams. Instead these sequences are gated by default and we look for
```

Comment:
descendants_are_gated could return true still for non-exam related reasons like pre-reqs. As far as for exams this ultimately calls down into edx-proctoring's time_limited_student_view which I'd expect to return None for exams not configured there. You bring up a good point and I think I need to revisit that assumption because that may have only been true for me locally. If it doesn't work that way now, we should be able to fix it to do that.

Reply:
@MichaelRoytman I found a cleaner way to handle this and just put a short circuit in seq_module before we call out to edx-proctoring. https://github.com/openedx/edx-platform/pull/31653/files#diff-8e54af31531a9368581fe6d673f8c82890db2326b25229b433716c1b0ba706bbR938